src/test/java/plugins/LdapPluginTest.java

```diff
@@ -63,7 +63,7 @@ private LdapDetails createDefaults(LdapContainer ldapContainer) {
      * @return default ldap connection details without the manager credentials
      */
     private LdapDetails createDefaultsWithoutManagerCred(LdapContainer ldapContainer) {
-        return new LdapDetails(ldapContainer.getHost(), ldapContainer.getPort(), "", "", ldapContainer.getRootDn());
+        return new LdapDetails(LdapContainer.addBracketsIfNeeded(ldapContainer.getHost()), ldapContainer.getPort(), "", "", ldapContainer.getRootDn());
```

Comment:
this appears incorrect.  LDAPDetails is expecting a host, `[22:33::12]` is not a host but an encapsulated ip6 host?

Reply:
Ditto https://github.com/jenkinsci/acceptance-test-harness/pull/1510#discussion_r1537489501

Reply:
Yes, but at some point this is used to create an URL which is not working without brackets on dual-stack mode

```
Caused: javax.naming.NamingException: Cannot parse url: ldap://2600:1900:4020:cd9b::6:389/dc=jenkins-ci,dc=org [Root exception is java.net.MalformedURLException: unsupported authority: 2600:1900:4020:cd9b::6:389]
	at java.naming/com.sun.jndi.ldap.LdapURL.<init>(LdapURL.java:115)
	at java.naming/com.sun.jndi.ldap.DefaultLdapDnsProvider.lookupEndpoints(DefaultLdapDnsProvider.java:48)
	at java.naming/com.sun.jndi.ldap.LdapDnsProviderService.lookupEndpoints(LdapDnsProviderService.java:109)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:166)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:247)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:154)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:84)
	at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:732)
	at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
	at java.naming/javax.naming.InitialContext.init(InitialContext.java:236)
	at java.naming/javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:154)
	at org.springframework.ldap.core.support.LdapContextSource.getDirContextInstance(LdapContextSource.java:42)
	at org.springframework.ldap.core.support.AbstractContextSource.createContext(AbstractContextSource.java:351)
```

This change makes it to work under IPv6 dual stack and IPv4